lang/cs/Org.Apache.REEF.IMRU.Tests/TestTaskManager.cs

```diff
@@ -0,0 +1,702 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on anAssert.Equal
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using NSubstitute;
+using Org.Apache.REEF.Driver.Context;
+using Org.Apache.REEF.Driver.Evaluator;
+using Org.Apache.REEF.Driver.Task;
+using Org.Apache.REEF.IMRU.OnREEF.Driver;
+using Org.Apache.REEF.IMRU.OnREEF.Driver.StateMachine;
+using Org.Apache.REEF.Network.Group.Driver;
+using Org.Apache.REEF.Tang.Implementations.Tang;
+using Org.Apache.REEF.Tang.Interface;
+using Org.Apache.REEF.Utilities;
+using Xunit;
+
+namespace Org.Apache.REEF.IMRU.Tests
+{
+    /// <summary>
+    /// Test cases for TaskManager
+    /// </summary>
+    public sealed class TestTaskManager
+    {
+        private const string MapperTaskIdPrefix = "MapperTaskIdPrefix";
+        private const string MasterTaskId = "MasterTaskId";
+        private const string EvaluatorIdPrefix = "EvaluatorId";
+        private const string ContextIdPrefix = "ContextId";
+        private const int TotalNumberOfTasks = 3;
+
+        /// <summary>
+        /// Tests valid Add task cases
+        /// </summary>
+        [Fact]
+        public void TestValidAdd()
```

Comment:
This test can be combined with `TestReset` (to make `TestAddAndReset`) if we add `Assert.Equal(TotalNumberOfTasks, taskManager.NumberOfTasks);` after `taskManager` initialization.

Also, can add validation that all tasks are created in `TaskState.TaskNew` state